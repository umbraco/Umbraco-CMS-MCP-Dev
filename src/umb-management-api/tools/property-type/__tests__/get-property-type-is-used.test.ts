import { getPropertyTypeIsUsedQueryParams } from "@/umb-management-api/umbracoManagementAPI.zod.js";
import GetPropertyTypeIsUsedTool from "../get/get-property-type-is-used.js";
import {
  createSnapshotResult,
  normalizeErrorResponse,
} from "@/test-helpers/create-snapshot-result.js";
import { jest } from "@jest/globals";
import { PropertyTypeTestHelper } from "./helpers/property-type-test-helper.js";

describe("get-property-type-is-used", () => {
  let originalConsoleError: typeof console.error;
  let headerControlsId: string | null;
  const titlePropertyAlias = "title";

  beforeEach(async () => {
    originalConsoleError = console.error;
    console.error = jest.fn();

    // Get the headerControls document type ID and title property alias
    headerControlsId = await PropertyTypeTestHelper.getHeaderControlsId();
  });

  afterEach(() => {
    console.error = originalConsoleError;
  });

  it("should check if a property type is used with both parameters", async () => {
    const params = getPropertyTypeIsUsedQueryParams.parse({
      contentTypeId: headerControlsId!,
      propertyAlias: titlePropertyAlias,
    });
    const result = await GetPropertyTypeIsUsedTool().handler(params, {
      signal: new AbortController().signal,
    });
    expect(createSnapshotResult(result)).toMatchSnapshot();
  });

  it("should handle empty parameters", async () => {
    const params = getPropertyTypeIsUsedQueryParams.parse({});
    const result = await GetPropertyTypeIsUsedTool().handler(params, {
      signal: new AbortController().signal,
    });
    expect(normalizeErrorResponse(result)).toMatchSnapshot();
  });
});
