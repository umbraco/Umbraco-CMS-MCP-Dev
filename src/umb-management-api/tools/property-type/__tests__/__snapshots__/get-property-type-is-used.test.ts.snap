// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-property-type-is-used should check if a property type is used with both parameters 1`] = `
{
  "content": [
    {
      "text": "true",
      "type": "text",
    },
  ],
}
`;

exports[`get-property-type-is-used should handle empty parameters 1`] = `
{
  "content": [
    {
      "text": "Error using get-property-type-is-used:
{
  "message": "Request failed with status code 400",
  "response": {
    "type": "https://tools.ietf.org/html/rfc9110#section-15.5.1",
    "title": "One or more validation errors occurred.",
    "status": 400,
    "errors": {
      "propertyAlias": [
        "The propertyAlias field is required."
      ]
    },
    "traceId": "normalized-trace-id"
  }
}",
      "type": "text",
    },
  ],
}
`;
